src/lib.rs

```diff
@@ -412,36 +595,39 @@ impl Table {
     }
 
     /// Directly inserts an [OwnedCapability] into this table.
-    pub fn insert_owned(&self, cap: OwnedCapability) -> Result<usize, TableError> {
+    pub fn import_owned(&self, cap: OwnedCapability) -> Result<CapabilityHandle, TableError> {
         assert_eq!(Arc::as_ptr(&self.post), Arc::as_ptr(&cap.post));
-        Ok(self.insert(cap.inner))
+        Ok(CapabilityHandle(self.insert(cap.inner).0))
     }
 
-    pub(crate) fn insert(&self, cap: Capability) -> usize {
-        self.inner.lock().insert(cap)
+    /// Helper function to directly insert a [Capability] into this table.
+    pub(crate) fn insert(&self, cap: Capability) -> CapabilityHandle {
```

Comment:
Hmmm, this is inconsistent with `TableInner::import` and `Table::import_owned` renaming it to `import` would conflict with the existing `import` method, which imports capabilities to mailbox routes.

Reply:
@airidaceae I have an idea: what if we created a `Mailbox::export` that exported an `OwnedCapability` that could be imported into tables using `Table::import_owned()`? That would break mailboxes out of the table API entirely and solve this issue.

Reply:
I like that. How does that interact with the requirement that the Table has the same post office as the mailbox? My concern is that a panic due to that issue may be pushed back pretty late into the importing process, but it does make each method do less things so I think its fine.

I guess its not up to the export method to decide know whats happening to capabilities after they're exported.

Reply:
This is no longer really a concern because that function returns an error now. Ill implement this now.